### PR TITLE
Added basic routes of customer producer and coffe, connected with database

### PR DESCRIPTION
**Pull request by Tezenn**
the functions are like on apiary. working

### CODE REVIEW

**Comment by leonhfr** [comment]
Awesome!
I have some ideas to make it even better:
- split the controller.js and model.js files in separate files (customer.controller.js, producer.controller.js, ...) to make it more maintainable
- with koa-router you can create and apply a new middleware like this `app.use(router.allowedMethods());` that automatically generates an answer to `OPTIONS` requests with the routes available as well as responding with `405 Method Not Allowed` and `501 Not Implemented` as appropriate.
- add an error handler in app.js, here is the one I used:

In `app.js`:
```js
const errorHandler    = require('./middlewares/error-handler');
app.use(errorHandler);
```
Error handler:
```js
'use strict';

const errorHandler = async (ctx, next) => {
  try {
    await next();
  } catch (err) {
    //eslint-disable-next-line
    console.error(err);
    ctx.body = undefined;
    ctx.status = ctx.status >= 400 && ctx.status || 400;
    if (err.message) {
      ctx.body = {
        errors: [err.message]
      };
    }
  }
};

module.exports = errorHandler;
```

**Review by leonhfr — APPROVED** [review]
Awesome!
I have some ideas to make it even better:
- split the controller.js and model.js files in separate files (customer.controller.js, producer.controller.js, ...) to make it more maintainable
- with koa-router you can create and apply a new middleware like this `app.use(router.allowedMethods());` that automatically generates an answer to `OPTIONS` requests with the routes available as well as responding with `405 Method Not Allowed` and `501 Not Implemented` as appropriate.
- add an error handler in app.js, here is the one I used:

In `app.js`:
```js
const errorHandler    = require('./middlewares/error-handler');
app.use(errorHandler);
```
Error handler:
```js
'use strict';

const errorHandler = async (ctx, next) => {
  try {
    await next();
  } catch (err) {
    //eslint-disable-next-line
    console.error(err);
    ctx.body = undefined;
    ctx.status = ctx.status >= 400 && ctx.status || 400;
    if (err.message) {
      ctx.body = {
        errors: [err.message]
      };
    }
  }
};

module.exports = errorHandler;
```

File: controllers/controller.js [hunk 1]
```diff
@@ -1,39 +1,75 @@
 const model = require('../models/model');
 
+// me //
+
+exports.getMe = async ctx => {
+  let id = ctx.request.header.authorization.split(' ')[1];
```
Comment (arol) [inline controllers/controller.js:6]: `const` 👉 10 push-ups

File: controllers/controller.js [hunk 2]
```diff
@@ -1,39 +1,75 @@
...
+  ctx.body = await model.getMe(id);
+  if (ctx.body) return (ctx.status = 200);
+  ctx.status = 404;
+};
+
+exports.updateMe = async ctx => {
+  let id = ctx.request.header.authorization.split(' ')[1];
```
Comment (arol) [inline controllers/controller.js:13]: Repeated code. Middleware?

File: controllers/controller.js [hunk 3]
```diff
@@ -1,39 +1,75 @@
...
 exports.getProducer = async ctx => {
   ctx.body = await model.getProducer(ctx.params.id);
-  ctx.status = 200;
+  if (ctx.body) return (ctx.status = 200);
+  ctx.status = 404;
+};
+
+exports.filterProducers = async ctx => {
+  let filter = ctx.params.filter;
+  let value = ctx.params.value;
```
Comment (arol) [inline controllers/controller.js:63]: All `consts`

File: models/model.js [hunk 4]
```diff
@@ -2,45 +2,123 @@ const Producer = require('../schemas/producer');
...
+      returning: true,
+      plain: true,
+      where: { id: id }
+    });
+    let customer = await Customer.find({
+      where: { id: id }
+    });
+    return customer;
+  }
+  /*   if (await Shipper.find({
```
Comment (arol) [inline models/model.js:64]: Remove commented code.

**Review by arol — COMMENTED** [review]
You're abusing of the use of `let`. Follow this rule: 

☝️ Everything is `const`. If you have to change it later you come back and replace the `const` for a `let`.

**Comment by arol** [comment]
@leonhfr good feedback but in terms of syntax, you're allowing all of this to go into production. I think this is a no-go.

**Pull request merged**